### ISSUE TITLE
XGBoost sched policy 58c: SCHED_MODEL_KIND_XGBOOST + sched_xgb.c + derived features

### ISSUE THREAD

**Issue by johnjezl** [SLM-OS/SLM-Operating-System#855]
## Description

Sub-ticket of #58. Wire the trained XGBoost cascade from #58a (#850) through the generalized traversal engine from #58b (#854) into a new pluggable scheduler policy registered alongside `ai_mlp`, `ai_ppo`, and `ai_hailo`.

### 1. Scheduler blob kind (sixth slot)

Add `SCHED_MODEL_KIND_XGBOOST = 6` to the scheduler blob kind enum. Touch points (greps already done):
- `kernel/include/sched_model.h` (or wherever the existing kind ids live)
- `kernel/src/lua_slm.c:2402` — `sched_model_kind_id()` name→id map
- `kernel/src/shell_sys.c:6421` — `sched_model_kind_id()` shell-side map
- `kernel/src/shell_sys.c:6431` — `sched_model_kind_name()` reverse map
- `kernel/src/blob_autoload.c` — autoload routing (existing `sched_blob_validate_file` / `sched_blob_stage_file` paths)
- `kernel/src/runtime_blob_file.c:190` — `sched_blob_stage_file` accepts the new kind

### 2. Rust predictor

New file `runtime/src/sched/xgb.rs`:
- Parses staged blob via the generalized `parse_payload` from #58b (cascade form, N=3).
- Stores three classifier graphs.
- Exposes `predict(state: &[f32; 108], derived: &[f32; 5]) -> (u8 core, u8 priority, u8 preempt)`:
  - Build per-classifier feature slice. `core_clf` consumes 113 features. `priority_clf` consumes 114 (113 + `core` as extra). `preempt_clf` consumes 115 (114 + `priority`). Match the Python cascade ordering exactly — see `~/projects/slm-os-scheduler-ai/training/xgboost/train.py:60 TripleClassifier.predict`.
  - Argmax over tree sums per classifier; map through `label_classes` from the blob's per-classifier label map.
- Round-trip test: load the `test_vectors_xgb.bin` from #58a, run predict on each, compare against `expected_actions_xgb.bin`. Must match bit-for-bit.

### 3. Derived features

Port the 5 derived features (`max_deadline_urgency`, `core_util_std`, `deadline_task_count`, `gpu_should_use`, `best_cache_fit_core`) from `~/projects/slm-os-scheduler-ai/training/xgboost/features.py` to either C (in `kernel/sched/ai/ai_state.c`) or Rust (in `runtime/src/sched/xgb.rs`). Recommend Rust-side so the predictor module is self-contained.

### 4. C policy vtable

New file `kernel/sched/ai/sched_xgb.c`. Pattern off `kernel/sched/ai/sched_ai.c`:

\`\`\`c
const struct sched_policy_ops sched_policy_ai_xgb = {
    .name = "ai_xgb",
    .init = sched_xgb_init,
    .shutdown = sched_xgb_shutdown,
    .assign_cpu = sched_xgb_assign_cpu,
    .tick = NULL,
    .has_gpu_backend = false,
};
\`\`\`

- `assign_cpu` extracts state via `ai_extract_state`, computes derived features, calls into Rust predictor via FFI (mirror the `ai_mlp_forward_logits` FFI pattern), decodes the returned `(core, priority, preempt)` triple, and returns the CPU id.
- **CPU-id clamping for Pi 5:** XGBoost was trained on 6 cores. Pi 5 has 4. When `core >= cpu_count`, clamp via `core % cpu_count` and fall through to heuristic; log once per session at WARN.
- Refuse to register if no XGBoost blob is active. Match how `sched_policy_ai_mlp` requires weights present.

### 5. Build wiring

- Add `AI_SCHED=ON` Makefile/CMake plumbing to compile `sched_xgb.c` and the Rust predictor.
- Register the policy in `sched_register_policy` startup order alongside MLP/PPO/Hailo.

### 6. Tests

- QEMU test (`make test`): policy registers, blob round-trips via the test framework's blob-stage helpers, predict produces deterministic decisions for a fixed input.
- Bit-equality test: 100 vectors from `test_vectors_xgb.bin` produce the actions in `expected_actions_xgb.bin`.

## Acceptance criteria

- `slm.sched_policy_list()` in Lua shows `ai_xgb` alongside other policies.
- `slm.sched_set_policy("ai_xgb")` succeeds after staging + activating the XGBoost blob via `slm.sched_model_status / activate`.
- Bit-equality test passes against Python ground truth.
- CPU-id clamping verified on Pi 5 builds.
- QEMU `make test` green; Pi 5 + Jetson hardware builds clean (real-hardware verification under #58d / #851).

## Out of scope

- Sibling-repo exporter → #58a (#850).
- Shared traversal engine → #58b (#854).
- Bench harness extension + docs → #58d (#851).

## Estimated Effort

~3 days.

## References

- Parent: #58. Umbrella: #449. Pluggable-policy framing: #848.
- Mirror policy: `kernel/sched/ai/sched_ai.c` (`sched_policy_ai_mlp` is the closest analogue).
